src/Stack/Options/ConfigParser.hs

```diff
@@ -128,6 +129,11 @@ configOptsParser currentDir hide0 =
             <> help "Install binaries to DIR"
             <> hide
              ))
+    <*> many (
+        strOption
+            ( long "setup-info-yaml"
+           <> help "Alternate URL or absolute path for stack dependencies"
```

Comment:
I would think a relative path would be even more convenient and recommended, so that the YAML file can be included in a repo and the code will work on multiple machines. Any underlying reason for preferring absolute paths?